<documentation title="Opening Function Brace Kerninghan Ritchie">
    <standard>
    <![CDATA[
    The function opening brace must be on the same line as the function declaration closing
    parenthesis, with exactly one space between them, and must be the last content on the line.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Opening brace on the same line.">
        <![CDATA[
function fooFunction($arg1, $arg2 = '')<em> {</em>
}
        ]]>
        </code>
        <code title="Invalid: Opening brace on the next line.">
        <![CDATA[
function fooFunction($arg1, $arg2 = '')
<em>{</em>
}
        ]]>
        </code>
    </code_comparison>
</documentation>
