<documentation title="Array Indent">
    <standard>
    <![CDATA[
    The opening brace of a multi-line array must be indented at least to the same level as the start of the statement.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Opening brace of a multi-line array indented at least to the same level as the start of the statement.">
        <![CDATA[
if ($condition) {
    $a =
<em>    [</em>
        1,
        2,
    ];
}

$b = <em>[</em>
    1,
    2,
];
        ]]>
        </code>
        <code title="Invalid: Opening brace of a multi-line array not indented at least to the same level as the start of the statement.">
        <![CDATA[
if ($condition) {
    $a =
<em>[</em>
        1,
        2,
    ];
}
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    The first line of each array element must be indented exactly four spaces more than the start of the statement.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: First line of each array element indented exactly four spaces more than the start of the statement.">
        <![CDATA[
$a = array(
<em>    </em>1,
<em>    </em>2,
<em>    </em>3,
);
        ]]>
        </code>
        <code title="Invalid: First line of each array element not indented exactly four spaces more than the start of the statement.">
        <![CDATA[
$a = array(
<em>  </em>1,
    2,
<em>        </em>3,
);
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    There must be a newline before the array closing brace.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Newline before the array closing brace.">
        <![CDATA[
$a = [
    1,
    2,<em>
]</em>;
        ]]>
        </code>
        <code title="Invalid: No newline before the array closing brace.">
        <![CDATA[
$a = [
    1,
    2,<em>]</em>;
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    The closing brace must be on the same level of indentation as the start of the statement containing the array opener.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Closing brace on the same level of indentation as the start of the statement containing the array opener.">
        <![CDATA[
$a = array(
    1,
    2,
<em>)</em>;
        ]]>
        </code>
        <code title="Invalid: Closing brace not on the same level of indentation as the start of the statement containing the array opener.">
        <![CDATA[
$a = array(
    1,
    2,
<em>  )</em>;
        ]]>
        </code>
    </code_comparison>
</documentation>
