<documentation title="Scope Closing Brace">
    <standard>
    <![CDATA[
    Indentation of a closing brace must match the indentation of the line containing the opening brace.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Closing brace aligned with opening brace.">
        <![CDATA[
function foo()
{
    echo 'opening brace at column 1';
    echo 'closing brace at column 1';
<em>}</em>

class Foo {
    echo 'class keyword starts at column 1';
    echo 'closing brace at column 1';
<em>}</em>

if ($a > $b) {
    echo 'conditional starts at column 1';
    echo 'closing brace at column 1';
<em>}</em>

<?php if ($something) { ?>
    <p>conditional opening tag at column 1</p>
    <p>conditional closing tag at column 1</p>
<?php } ?>
        ]]>
        </code>
        <code title="Invalid: Closing brace misaligned with opening brace.">
        <![CDATA[
function foo()
{
    echo 'opening brace at column 1';
    echo 'closing brace at column 2';
<em> }</em>

class Foo {
    echo 'class keyword starts at column 1';
    echo 'closing brace at column 2';
<em> }</em>

if ($a > $b) {
    echo 'conditional starts at column 1';
    echo 'closing brace at column 5';
<em>    }</em>

<?php if ($something) { ?>
    <p>conditional opening tag at column 1</p>
    <p>conditional closing tag at column 2</p>
 <?php } ?>
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    Closing brace must be on a line by itself.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Close brace on its own line.">
        <![CDATA[
enum Foo {
<em>}</em>
        ]]>
        </code>
        <code title="Invalid: Close brace on a line containing other code.">
        <![CDATA[
enum Foo {<em>}</em>
        ]]>
        </code>
    </code_comparison>
</documentation>
