<documentation title="Scope Closing Brace">
    <standard>
    <![CDATA[
    Indentation of a closing brace must match the indentation of the line containing the opening brace.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Closing brace aligned with opening brace.">
        <![CDATA[
function foo()
{
    echo 'opening brace at column 1';
    echo 'closing brace at column 1';
<em>}</em>
        ]]>
        </code>
        <code title="Invalid: Closing brace misaligned with opening brace.">
        <![CDATA[
function foo()
{
    echo 'opening brace at column 1';
    echo 'closing brace at column 2';
<em> }</em>
        ]]>
        </code>
    </code_comparison>
    <code_comparison>
        <code title="Valid: Closing brace aligned with class keyword.">
        <![CDATA[
class Foo {
    echo 'class keyword starts at column 1';
    echo 'closing brace at column 1';
<em>}</em>
        ]]>
        </code>
        <code title="Invalid: Closing brace misaligned with class keyword.">
        <![CDATA[
class Foo {
    echo 'class keyword starts at column 1';
    echo 'closing brace at column 2';
<em> }</em>
        ]]>
        </code>
    </code_comparison>
    <code_comparison>
        <code title="Valid: Closing brace aligned with if keyword.">
        <![CDATA[
if ($a > $b) {
    echo 'conditional starts at column 1';
    echo 'closing brace at column 1';
<em>}</em>
        ]]>
        </code>
        <code title="Invalid: Closing brace aligned with content.">
        <![CDATA[
if ($a > $b) {
    echo 'conditional starts at column 1';
    echo 'closing brace at column 5';
<em>    }</em>
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    Closing brace must be on a line by itself.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Newline before closing brace.">
        <![CDATA[
enum Foo {
<em>}</em>
        ]]>
        </code>
        <code title="Invalid: No newline before closing brace.">
        <![CDATA[
enum Foo {<em>}</em>
        ]]>
        </code>
    </code_comparison>
</documentation>
