<documentation title="Increment Decrement Spacing">
    <standard>
    <![CDATA[
    There should be no whitespaces between a pre-increment/pre-decrement operator and a variable.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: No whitespaces between a pre-increment/pre-decrement operator and a variable.">
        <![CDATA[
++<em></em>$i;
--<em></em>ClassName::$prop;
--<em></em>$obj->prop;
        ]]>
        </code>
        <code title="Invalid: Whitespaces found between a pre-increment/pre-decrement operator and a variable.">
        <![CDATA[
++<em> </em>$i;
--<em>    </em>ClassName::$prop;
--<em>
</em>$obj->prop;
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    There should be no whitespaces between a variable and a post-increment/post-decrement operator.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: No whitespaces between a variable and a post-increment/post-decrement operator.">
        <![CDATA[
<em></em>$i++;
<em></em>ClassName::$prop++;
<em></em>$obj->prop--;
        ]]>
        </code>
        <code title="Invalid: Whitespaces found between a variable and a post-increment/post-decrement operator.">
        <![CDATA[
$i<em> </em>++;
ClassName::$prop<em>    </em>++;
$obj->prop<em>
</em>--;
        ]]>
        </code>
    </code_comparison>
</documentation>
