<documentation title="Language Construct Spacing">
    <standard>
    <![CDATA[
    Language constructs that can be used without parentheses must contain a single space between themselves and their content.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Single space after language construct.">
        <![CDATA[
echo<em> </em>'Hello, World!';
throw<em> </em>new<em> </em>Exception();
return<em> </em>$newLine;
        ]]>
        </code>
        <code title="Invalid: No space, more than one space or newline after language construct.">
        <![CDATA[
echo<em></em>'Hello, World!';
throw<em>   </em>new<em>   </em>Exception();
return<em></em>
$newLine;
        ]]>
        </code>
    </code_comparison>
</documentation>
