<documentation title="Switch Declaration">
    <standard>
    <![CDATA[
    Case statements must be followed by exactly one space.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Case statement followed by one space.">
        <![CDATA[
switch ($foo) {
    case<em> </em>'bar':
        break;
}
        ]]>
        </code>
        <code title="Invalid: Case statement not followed by one space.">
        <![CDATA[
switch ($foo) {
    case<em></em>'bar':
        break;
}
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    There must be no whitespace between the case value or default keyword and the colon.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Colons not preceded by whitespace.">
        <![CDATA[
switch ($foo) {
    case 'bar'<em></em>:
        break;
    default<em></em>:
        break;
}
        ]]>
        </code>
        <code title="Invalid: Colons preceded by whitespace.">
        <![CDATA[
switch ($foo) {
    case 'bar'<em> </em>:
        break;
    default<em> </em>:
        break;
}
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    Terminating statements must be indented four more spaces from the case statement.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Break statement indented correctly.">
        <![CDATA[
switch ($foo) {
    case 'bar':
<em>        </em>break;
}
        ]]>
        </code>
        <code title="Invalid: Break statement not indented four spaces.">
        <![CDATA[
switch ($foo) {
    case 'bar':
<em>    </em>break;
}
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    There must be a comment when fall-through is intentional in a non-empty case body.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Comment marking intentional fall-through in a non-empty case body.">
        <![CDATA[
switch ($foo) {
    case 'bar':
        echo $foo;
        <em>// no break</em>
    default:
        break;
}
        ]]>
        </code>
        <code title="Invalid: No comment marking intentional fall-through in a non-empty case body.">
        <![CDATA[
switch ($foo) {
    case 'bar':
        echo $foo;<em></em>
    default:
        break;
}
        ]]>
        </code>
    </code_comparison>
</documentation>
